src/main/java/org/jenkinsci/plugins/mesos/MesosComputer.java

```diff
@@ -45,4 +55,19 @@ public void taskCompletedWithProblems(
   public String toString() {
     return String.format("%s (slave: %s)", getName(), getNode());
   }
+
+  @Override
+  public MesosSlave getNode() {
+    return super.getNode();
+  }
+
+  @Override
+  public HttpResponse doDoDelete() throws IOException {
+    try {
+      getNode().terminate();
+    } catch (InterruptedException e) {
+      logger.warn(" got exception " + e + "failure to delete agent" + getNode().getPodId());
```

Comment:
```suggestion
      logger.warn("Failure to terminate agent {}", getNode().getPodId(), e);
```

That is ok for now. Once we decided on a crash strategy we can address it again. I would simply throw an unchecked `MesosPluginCrashException(e)`